dali/test/python/test_operator_segmentation_random_mask_pixel.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+from test_utils import check_batch, dali_type
+import random
+from segmentation_test_utils import make_batch_select_masks
+from nose.tools import assert_raises
+
+np.random.seed(4321)
+
+def check_random_mask_pixel(ndim=2, batch_size=3,
+                            min_extent=20, max_extent=50):
+    pipe = dali.pipeline.Pipeline(batch_size=batch_size, num_threads=4, device_id=0, seed=1234)
+    with pipe:
+        # Input mask
+        in_shape_dims = [fn.cast(fn.uniform(range=(min_extent, max_extent + 1), shape=(1,), device='cpu'),
+                                 dtype=types.INT32) for d in range(ndim)]
+        in_shape = fn.cat(*in_shape_dims, axis=0)
+        in_mask = fn.cast(fn.uniform(range=(0, 2), device='cpu', shape=in_shape), dtype=types.INT32)
+
+        fg_pixel1 = fn.segmentation.random_mask_pixel(in_mask, foreground=1)  # > 0
+        fg_pixel2 = fn.segmentation.random_mask_pixel(in_mask, foreground=1, threshold=0.99)  # > 0.99
+        fg_pixel3 = fn.segmentation.random_mask_pixel(in_mask, foreground=1, value=2)  # == 2
+        rnd_pixel = fn.segmentation.random_mask_pixel(in_mask, foreground=0)
+        coin_flip = fn.coin_flip(probability=0.7)
+        fg_biased = fn.segmentation.random_mask_pixel(in_mask, foreground=coin_flip)
+
+        # Demo purposes: Taking a random pixel and produce a valid anchor to feed slice
+        crop_shape = in_shape - 2  # We want to force the center adjustment, therefore the large crop shape
+        anchor = fg_pixel1 - crop_shape // 2
+        anchor = min(max(0, anchor), in_shape - crop_shape)
```

Comment:
So it is up to the user to make sure that window is inside the image. Maybe it could be mentioned in the documentation,

Reply:
Right now this operator is not advertised to be used for cropping. It's a general purpose pixel selector. Do you think we should add this suggestion of usage in the operator documentation?

Reply:
Maybe not if we are not advertising it that way.

Reply:
I think we could add an example using this operator to calculate a crop center, maybe in one of the existing notebooks